README.md

```diff
@@ -1,20 +1,88 @@
+|    L.1    |        Live Spork support for registers       |         X        |         X         |         X        |                   |    X    |      X     |      X      |        X       |                |                |                   |                    |              |
+|    L.2    |         Live Spork support with events        |         X        |         X         |         X        |         X         |    X    |      X     |      X      |        X       |                |                |                   |                    |              |
+|    R.2    | Rosetta API support with sub-account balances |         X        |         X         |         X        |         X         |    X    |      X     |      X      |        X       |        X       |        X       |         X         |                    |              |
+|    C.1    |   Cryptographic Verification of local state   |         X        |         X         |         X        |         X         |    X    |      X     |      X      |        X       |        X       |        X       |         X         |          X         |              |
+|    C.2    |  Cryptographic Proofs for remote state access |         X        |         X         |         X        |         X         |    X    |      X     |      X      |        X       |        X       |        X       |         X         |          X         |       X      |
+
+## Architecture
+
+### Components
+
+The Flow Data Provisioning Service is constituted of three main components.
+
+The **indexer** is at the core of the service. It indexes deltas into its own data structures and provides a low level
+API to expose its state.
+The **mapper** synchronizes the **indexer** with a source of deltas (`FilesystemChain` or `NetworkChain`.)
+The **chain** is in charge of fetching block sequence data (from a WAL in the filesystem for past sporks, or a network
+socket for live ones.)
 
-- index transaction events
-- events retrieval interface
-- REST API interface
-- DPS web client
+### Diagram
 
-## Interfaces
+The following diagram describes a simple overview of an example where the DPS is reading from two sporks. One of them
+is a live one, which sends
 
-1. Indexer: indexes changes/deltas per block
-2. Update Stream: gets all the trie updates per chunk
-3. Chain View: has block sequence data
\ No newline at end of file
+```text
+┌─────────────────┐
+│                 │
+│   Past Spork    │
+│                 │
+│  ┌───────────┐  │             ┌───────┐
+│  │           │  │             │       │
+│  │ Exec Node ├──┼────────────►│ WALog │
+│  │           │  │             │       │
+│  └───────────┘  │             └───┬───┘
+│                 │                 │
+│                 │                 │
+└─────────────────┘                 │
```

Comment:
The first step in this part of the diagram would be `LedgerWAL` and `Badger DB`. `LedgerWAL` would feed into the `WALStreamer`, and `Badger DB` would feed into the `FSChain`.

Reply:
I'm not sure what you mean exactly? Would it be the following

```
LedgerWAL -> WALStreamer 
BadgerDB -> FSChain -> WALStreamer
WALStreamer -> Index
```

What would feed into BadgerDB?

Reply:
Also shouldn't it be called WALMapper instead of streamer, to stay consistent with the LiveMapper and the components?

Reply:
BadgerDB is a source, not a destination.